src/representations/calculator_spherical_covariants.hh

```diff
@@ -53,138 +53,105 @@ namespace rascal {
   namespace internal {
     enum class SphericalCovariantsType { LambdaSpectrum, End_ };
 
-    template <SphericalCovariantsType SpectrumType>
-    struct SphericalCovariantsPrecomputation {};
-
-    template <>
-    struct SphericalCovariantsPrecomputation<
-        SphericalCovariantsType::LambdaSpectrum>
-        : SphericalInvariantsPrecomputationBase {
-      using Parent = SphericalInvariantsPrecomputationBase;
-      using Hypers_t = typename SphericalInvariantsPrecomputationBase::Hypers_t;
-
-      explicit SphericalCovariantsPrecomputation(const Hypers_t & hypers) {
-        this->max_angular = hypers.at("max_angular").get<size_t>();
-        this->inversion_symmetry = hypers.at("inversion_symmetry").get<bool>();
-        this->lambda = hypers.at("lam").get<size_t>();
-
-        // get the number of non zero elements in the w3j
-        // Use the well-known selection rules for the Wigner 3j symbols
-        int n_elements{0};
-        size_t l3{this->lambda};
-        for (size_t l1{0}; l1 < this->max_angular + 1; ++l1) {
-          for (size_t l2{0}; l2 < this->max_angular + 1; ++l2) {
-            // Triangle rule
-            if (l1 < static_cast<size_t>(std::abs<int>(l2 - l3)) ||
-                l1 > l2 + l3) {
+    inline Eigen::ArrayXd precompute_wigner_3js(size_t max_angular,
```

Comment:
Is the `inline` really useful? I know it's been peppered throughout the code, but I thought compilers mostly ignored inlining hints nowadays.

Reply:
Here yes, but not for code inlining. `inline` is a way around the One Definition Rule. Since the file is included and compiled in multiple difference compilation units, the linker then have multiples copies of the function to work with. `inline` tells the linker they are all equivalent and that it should use any of them.

Reply:
I removed most of the other inline qualifiers in #150

Reply:
ergh, ok, leave it to C++ to give you a keyword that does something only vaguely related to what it says and requires an advanced degree in compiler design to actually understand.